auslib/blobs/whitelist.py

```diff
@@ -0,0 +1,42 @@
+import re
+from auslib.blobs.base import Blob
+
+import logging
+log = logging.getLogger(__name__)
+
+INDENT = 4 * ' '
+SETTING_TMPL = ('<setting id="%(id)s" '
+                'lastModified="%(last_modified)s"/>')
+
+
+class WhitelistBlobV1(Blob):
+    format_ = {
+        'name': None,
+        'schema_version': None,
+        'whitelist': [
+            { 'imei': None }
+        ]
+    }
+
+    def __init__(self, **kwargs):
+        Blob.__init__(self, **kwargs)
+        if "schema_version" not in self:
+            self["schema_version"] = 3000
+
+    def isWhitelisted(self, requestIMEI):
+        self.log.debug("Checking to see if IMEI '%s' is whitelisted", requestIMEI)
+        if any(listItem['imei'] == requestIMEI for listItem in self['whitelist']):
+            self.log.debug("IMEI is whitelisted")
+            return True
+        self.log.debug("IMEI not whitelisted")
+        return False
+
+    def shouldServeUpdate(self, updateQuery):
+        self.log.debug(updateQuery)
+        requestIMEI = updateQuery['IMEI']
```

Comment:
One thing to be careful of here: not every update query type will have an IMEI present. Probably safer to use updateQuery.get() here...